opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpanSource.java

```diff
@@ -36,7 +34,12 @@ public ThreadLocalActiveSpan activeSpan() {
 
     @Override
     public ActiveSpan makeActive(Span span) {
-        return new ThreadLocalActiveSpan(this, span, new AtomicInteger(1));
+        return new ThreadLocalActiveSpan(this, span, new AutoFinisher());
```

Comment:
Its a difficult decision - less breaking is easier in the short term, but doesn't truly decouple the active span lifecycle from the ref counting concern. Although it will impact all instrumentations that have used active spans, I think I would prefer to have the instrumentations explicitly request ref counting via setting the `AutoFinisher` observer, else handle finishing themselves.